Objects/longobject.c

```diff
@@ -4492,6 +4492,12 @@ long_rshift1(PyLongObject *a, Py_ssize_t wordshift, digit remshift)
     PyLongObject *z = NULL;
     Py_ssize_t newsize, hishift, i, j;
     twodigits accum;
+    digit sticky;
+
+    /* Special-case a shift of zero. */
+    if (wordshift == 0 && remshift == 0) {
```

Comment:
It may be better to special-case this earlier, _before_ we've gone to the effort of computing `wordshift` and `remshift`.

Reply:
It is already special-case `Py_SIZE(a) == 0` before calling `long_rshift1()`.

Reply:
Ah, we need to special-case also `Py_SIZE(b) == 0`. But this is an uncommon case, so it does not matter.

Reply:
Yes, I'm not sure whether it's worth special-casing `shift == 0` in general, or whether to just move that handling into the first `if (a_negative)` branch.

Reply:
Changed to handle the `shift == 0` case only when we need to, instead of special-casing it early.